### PR TITLE
E-830 : Allow users to claim a project

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: src/api/me/mutations.ts [hunk 1]
```diff
@@ -18,4 +18,20 @@ const useAcceptProjectLeaderInvitation = ({
   });
 };
 
-export default { useAcceptProjectLeaderInvitation };
+const useClaimProject = ({
+  params,
+  options = {},
+}: UseMutationProps<unknown, { projectId?: string; projectSlug?: string }, unknown>) => {
+  return useBaseMutation<unknown, unknown>({
```
Comment (haydencleary) [inline src/api/me/mutations.ts:25]: Don't we know the return type ?

File: src/pages/ProjectDetails/Banners/ClaimBanner/ClaimBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      return false;
+    }
+
+    const isAllOrganizationInstalled = project?.organizations?.every(org => org.installed);
+
+    if (isAllOrganizationInstalled) {
+      return true;
+    }
+
+    return false;
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Banners/ClaimBanner/ClaimBanner.tsx:74]:
```suggestion
    return isAllOrganizationInstalled;
```

File: src/pages/ProjectDetails/Banners/ClaimBanner/ClaimBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  };
+
+  const onSubmitClaim = () => {
+    if (canSubmit) {
+      claimProjectMutation({});
+    }
+  };
+
+  useEffect(() => {
+    if (searchParams && searchParams.get("claim_callback")) {
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Banners/ClaimBanner/ClaimBanner.tsx:96]:
```suggestion
    if (searchParams?.get("claim_callback")) {
```

File: src/pages/ProjectDetails/Banners/ClaimBanner/ClaimBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+              {project?.organizations?.map(org => (
+                <ClaimBannerOrganization
+                  key={org.id}
+                  organization={org}
+                  myOrganizations={myOrganizations || []}
+                  project={project}
+                />
+              ))}
+            </div>
+            <div className=" mt-6">
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Banners/ClaimBanner/ClaimBanner.tsx:143]:
```suggestion
            <div className="mt-6">
```

File: src/translations/en.json [hunk 5]
```diff
@@ -774,6 +774,26 @@
   },
   "project": {
     "stillFetching": "We are still fetching some project information. Please come back in approx. 5-10 min.",
+    "claim": {
+      "banner": {
+        "content": "Working at cal.com? You’re 3 click away to claim access to your own project on Onlydust!",
```
Comment (haydencleary) [inline src/translations/en.json:779]: Isn't `cal.com` hardcoded ? Also should be `3 clicks away`

File: src/utils/githubSetupLink.ts [hunk 6]
```diff
@@ -13,6 +14,10 @@ export const OAuthGithubConfigLink = `https://github.com/settings/connections/ap
   import.meta.env.VITE_GITHUB_OAUTH_APP_ID
 }`;
 
+export const GithubState = {
+  claim: "claim-state-",
+  edit: "claim-state-",
```
Comment (haydencleary) [inline src/utils/githubSetupLink.ts:19]: Normal that these are the same ?

Comment (alexbeno) [reply src/utils/githubSetupLink.ts:19]: Thanks :p

**Review by pixelfact — APPROVED** [review]

File: src/translations/en.json (hunk 5 above)
Comment (alexbeno) [reply src/translations/en.json:779]: Thanks !

Comment (alexbeno) [reply src/translations/en.json:779]: 3 clicks away ?

File: src/api/me/mutations.ts (hunk 1 above)
Comment (alexbeno) [reply src/api/me/mutations.ts:25]: it's a put so we don't have any response from the backend, juste a success code

**Pull request merged**